### DISCUSSION TITLE
designspacelib: axis minimum and maximum values in designspace coordinate?

### DISCUSSION THREAD

**Discussion by LettError** [fonttools/fonttools#3257] [Q&A] [answered]
Is there a recommended way to get the extreme values of an axis in designspace coordinates?
Usecase: I need to clip designspace coordinates to axis extremes. 
@madig

**Comment by justvanrossum** [comment, marked as answer]
Axes have a `map_forward()` method, which maps values from user space to designspace. So feed it the `axis.minimum` and `axis.maximum` and Bob's your uncle:

```python
from fontTools.designspaceLib import DesignSpaceDocument

p = "path/to/Font.designspace"

doc = DesignSpaceDocument.fromfile(p)
for axis in doc.axes:
    print(axis.name)
    print(axis.minimum, axis.map_forward(axis.minimum))
    print(axis.maximum, axis.map_forward(axis.maximum))
```

**Comment by LettError** [comment]
One follow up: are there known use cases for mapping applied to discrete axis values?
